Jenkinsfile

```diff
@@ -0,0 +1,116 @@
+pipeline {
+  
+  agent {
+    docker {
+      image 'node:6.9.5'
+      args "-v /var/lib/jenkins/.npm:/tmp/.npm"
+    }
+  }
+  
+  environment  {
+      HOME = "/tmp"
+  }
+  
+  triggers {
+    pollSCM('* * * * *')
+    cron('@daily')
+  }
+  
+  stages {
+    stage('Set up') {
+      steps {
+        // we need to disable logallrefupdates, else git clones during the npm install will require git to lookup the user id
+        // which does not exist in the container's /etc/passwd file, causing the clone to fail.
+        sh 'git config --global core.logallrefupdates false'
+        
+        sh 'rm -rf node_modules/*'
+      }
+    }
+    
+    stage('Clone Dependencies') {
+      steps {
+        sh 'rm -rf public/brand modules'
+        checkout([$class: 'GitSCM', branches: [[name: '*/master']], extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'public/brand'], [$class: 'CloneOption', shallow: true]], userRemoteConfigs: [[credentialsId: 'GIT_DEPLOY_KEY', url: 'git@github.com:sharelatex/brand-sharelatex']]])
```

Comment:
Are all of these `[]`s necessary? It looks like a lot can be optional: https://jenkins.io/doc/book/pipeline/jenkinsfile/#optional-step-arguments